pandas/core/arrays/categorical.py

```diff
@@ -2699,10 +2699,9 @@ def _get_codes_for_values(values, categories: Index) -> np.ndarray:
     """
     dtype_equal = is_dtype_equal(values.dtype, categories.dtype)
 
-    if is_extension_array_dtype(categories.dtype) and is_object_dtype(values):
+    if isinstance(categories.dtype, ExtensionDtype) and is_object_dtype(values):
         # Support inferring the correct extension dtype from an array of
         # scalar objects. e.g.
-        # Categorical(array[Period, Period], categories=PeriodIndex(...))
```

Comment:
was this removal intentional?

Reply:
yes

Reply:
probably should remove the `e.g.` on the previous line as well?

Reply:
Sorry, I meant it was not intentional

Reply:
I see it has been restored in the latest commit. marking as resolved.

Reply:
Finally green :)